magefile.go

```diff
@@ -1295,3 +1320,170 @@ func shouldBuildAgent() bool {
 	}
 	return ret
 }
+
+// Pre-requisite: user must have the gcloud CLI installed
+func authGCP(ctx context.Context) error {
+	// Check if gcloud CLI is installed
+	const cliName = "gcloud"
+	cmd := exec.CommandContext(ctx, "which", cliName)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("%s CLI is not installed: %w", cliName, err)
+	}
+
+	// Check if user is already authenticated
+	var authList []struct {
+		Account string `json:"account"`
+	}
+	for authSuccess := false; !authSuccess; {
+		cmd = exec.CommandContext(ctx, cliName, "auth", "list", "--filter=status:ACTIVE", "--format=json")
+		output, err := cmd.Output()
+		if err != nil {
+			return fmt.Errorf("unable to list authenticated accounts: %w", err)
+		}
+
+		if err := json.Unmarshal(output, &authList); err != nil {
+			return fmt.Errorf("unable to parse authenticated accounts: %w", err)
+		}
+
+		if len(authList) > 0 {
+			// We have at least one authenticated, active account. All set!
+			authSuccess = true
+			continue
+		}
+
+		fmt.Fprintln(os.Stderr, "❌  GCP authentication unsuccessful. Retrying...")
+
+		// Try to authenticate user
+		cmd = exec.CommandContext(ctx, cliName, "auth", "login")
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("unable to authenticate user: %w", cliName, err)
+		}
+	}
+
+	// Skip additional Elastic-specific checks if so requested
+	const skipEnvVar = "TEST_INTEG_AUTH_GCP_SKIP_ELASTIC_CHECK"
```

Comment:
Maybe we can use an env var or an argument to the mage target to specify a a project to switch to instead of disabling the check on the email and project switching...
This should also make it easy to create different launch configs for IDEs etc.

Reply:
I like this idea - will do!

Reply:
Done in caeecb1ec.